## Load adversary

In [8]:
import numpy as np
from tensorflow import keras

adversary = keras.models.load_model('models/adv_new_0.65_test_r2')

## Create (target) model

In [9]:
from pia_functions import train_gradient_boosting_shadow_model, data_train_test
import pandas as pd

In [10]:
# load data
X_train, X_test, y_train, y_test, sensitive, sensitive_t = data_train_test()
# create model
gb = train_gradient_boosting_shadow_model(X_train, y_train, 0)

## Generate model output

In [11]:
# load synthetic model input
model_input = pd.read_csv('data/syn_data-new.csv')
# generate model output
out = gb.predict_proba(model_input)
# reformat model output
adversary_input = np.expand_dims(out[:,0], axis=0)

## Attack

In [12]:
# generate adversary prediction
adversary.predict(adversary_input)

1/1 [==============================] - 0s 81ms/step


array([[0.8010835]], dtype=float32)

Result interpretation: The adversary has inferred that 80.1% of the target model's training data are male samples. 